4/‎m_geomaxlite.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2024 reverse
+ *
+ * This file contains a third-party module for InspIRCd. You can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/// $ModAuthor: reverse <mike.chevronnet@gmail.com>
+/// $ModDesc: Adds city and country information to WHOIS using local MaxMind database and with it's own usermode +y.
+/// $ModConfig: <geolite dbpath="path/geodata/GeoLite2-City.mmdb">
+/// $ModDepends: core 4
+
+/// $LinkerFlags: -lmaxminddb
+
+/// $PackageInfo: require_system("alpine") libmaxminddb-dev pkgconf
+/// $PackageInfo: require_system("arch") pkgconf libmaxminddb
+/// $PackageInfo: require_system("darwin") libmaxminddb pkg-config
+/// $PackageInfo: require_system("debian~") libmaxminddb-dev pkg-config
+/// $PackageInfo: require_system("rhel~") pkg-config libmaxminddb-devel
+
+#include "inspircd.h"
+#include "modules/whois.h"
+#include "extension.h"
+#include <maxminddb.h>
+
+class GeoLiteMode final : public SimpleUserMode
```

Comment:
If you're not going to customise the mode behaviour there's no need to subclass it just use `SimpleUserMode` directly.